src/renderer/components/+helm-releases/release-details/release-details-model/release-details-model.injectable.tsx

```diff
@@ -0,0 +1,312 @@
+import type { CallForHelmReleaseUpdate } from "../../update-release/call-for-helm-release-update/call-for-helm-release-update.injectable";
+import updateReleaseInjectable from "../../update-release/update-release.injectable";
+import type { ShowCheckedErrorNotification } from "../../../notifications/show-checked-error.injectable";
+import showCheckedErrorNotificationInjectable from "../../../notifications/show-checked-error.injectable";
+import type { ShowNotification } from "../../../notifications";
+import showSuccessNotificationInjectable from "../../../notifications/show-success-notification.injectable";
+import React from "react";
+import createUpgradeChartTabInjectable from "../../../dock/upgrade-chart/create-upgrade-chart-tab.injectable";
+import type { HelmRelease } from "../../../../../common/k8s-api/endpoints/helm-releases.api";
+import type { NavigateToHelmReleases } from "../../../../../common/front-end-routing/routes/cluster/helm/releases/navigate-to-helm-releases.injectable";
+import navigateToHelmReleasesInjectable from "../../../../../common/front-end-routing/routes/cluster/helm/releases/navigate-to-helm-releases.injectable";
+import assert from "assert";
+import withOrphanPromiseInjectable from "../../../../../common/utils/with-orphan-promise/with-orphan-promise.injectable";
+
+const releaseDetailsModelInjectable = getInjectable({
+  id: "release-details-model",
+
+  instantiate: (di, targetRelease: TargetHelmRelease) => {
+    const callForHelmRelease = di.inject(callForHelmReleaseInjectable);
+    const callForHelmReleaseConfiguration = di.inject(callForHelmReleaseConfigurationInjectable);
+    const themeStore = di.inject(themeStoreInjectable);
+    const getResourceDetailsUrl = di.inject(getResourceDetailsUrlInjectable);
+    const updateRelease = di.inject(updateReleaseInjectable);
+    const showCheckedErrorNotification = di.inject(showCheckedErrorNotificationInjectable);
+    const showSuccessNotification = di.inject(showSuccessNotificationInjectable);
+    const createUpgradeChartTab = di.inject(createUpgradeChartTabInjectable);
+    const navigateToHelmReleases = di.inject(navigateToHelmReleasesInjectable);
+    const withOrphanPromise = di.inject(withOrphanPromiseInjectable);
+
+    const model = new ReleaseDetailsModel({
+      callForHelmRelease,
+      targetRelease,
+      themeStore,
+      callForHelmReleaseConfiguration,
+      getResourceDetailsUrl,
+      updateRelease,
+      showCheckedErrorNotification,
+      showSuccessNotification,
+      createUpgradeChartTab,
+      navigateToHelmReleases,
+    });
+
+    const load = withOrphanPromise(model.load);
+
+    // TODO: Reorganize Drawer to allow setting of header-bar in children to make "getPlaceholder" from injectable usable.
+    load();
+
+    return model;
+  },
+
+  lifecycle: lifecycleEnum.keyedSingleton({
+    getInstanceKey: (di, release: TargetHelmRelease) =>
+      `${release?.namespace}/${release?.name}`,
+  }),
```

Comment:
This seems weird, why is it `release?.namespace` and not `release.namespace`? If the reason is that `release` could be `undefined | null` then why is the function not declaring that?

Reply:
Good catch. Removed the optional chaining.